### PR TITLE
Block alignment - remove unnecessary class application for dynamic blocks.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
## Description

As a follow up to #24122 - some existing dynamic blocks using the `align` hook now have unnecessary code to apply class names for block alignment.  This PR removes these now obsolete bits for the following blocks:
* archives
* calendar
* categories
* latest-comments
* latest-posts
* navigation
* rss
* search
* tag-cloud

## How has this been tested?

Tested on local docker environment.

Verify the listed blocks properly have their block align classnames added to the front-end when applicable.
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/archives/index.php [hunk 1]
```diff
@@ -19,10 +19,6 @@ function render_block_core_archives( $attributes ) {
```
Comment (Addison-Stavlo) [inline packages/block-library/src/archives/index.php:19]: A bit of a side note - I noticed that many of these blocks still have `align` defined in their attributes along with an "enum" array:
https://github.com/WordPress/gutenberg/blob/0d08cdab0c2aed8611ad4468b86e29247590c60b/packages/block-library/src/archives/block.json#L5-L14

I would have thought defining `align` in the attributes to no longer be necessary as the hook does this for us.  However, removing this will result in an `Invalid parameter(s): attributes` error for already existing blocks so I have not removed any of this.

However, that does make me wonder if this `enum` param is necessary in all cases and if it should be added to the align hook for defining the attribute when it does not exist? :

https://github.com/WordPress/gutenberg/blob/0d08cdab0c2aed8611ad4468b86e29247590c60b/packages/block-editor/src/hooks/align.js#L95-L97

Comment (Copons) [reply packages/block-library/src/archives/index.php:19]: Could it be that blocks throwing errors just need to deprecate the old attributes set?
Something like this:
```jsx
const settings = {
  attributes: {
    foo: { type: 'string' },
  },
  deprecated: [
    {
      attributes: {
        align: { type: 'string', enum: [ /* ... */ ] },
        foo: { type: 'string' },
      },
    }
  ],
};
```

Comment (Addison-Stavlo) [reply packages/block-library/src/archives/index.php:19]: Oh probably, Im more concerned if there was any other reason for keeping this definition (and not deprecating) when the `align` hook was originally added to these blocks.  And if that `enum` should be part of the hook itself.

Comment (youknowriad) [reply packages/block-library/src/archives/index.php:19]: I feel the enum makes sense for all blocks. I think it adds some level of validation.

Comment (youknowriad) [reply packages/block-library/src/archives/index.php:19]: and yes, removing the definition of these attributes also make sense.

File: packages/block-library/src/archives/index.js [hunk 2]
```diff
@@ -20,4 +21,5 @@ export const settings = {
 	icon,
 	example: {},
 	edit,
+	deprecated,
```
Comment (youknowriad) [inline packages/block-library/src/archives/index.js:24]: I'm not sure I understand why we need "deprecated" here at all?

Comment (Copons) [reply packages/block-library/src/archives/index.js]: My bad there, I suggested it because I've misunderstood the error that was happening, which is caused by the block renderer endpoint instead.
I'll leave to @Addison-Stavlo to explain what's going on! 🙂

File: packages/block-library/src/archives/edit.js [hunk 3]
```diff
@@ -36,7 +40,7 @@ export default function ArchivesEdit( { attributes, setAttributes } ) {
 			<Disabled>
 				<ServerSideRender
 					block="core/archives"
-					attributes={ attributes }
+					attributes={ omit( attributes, [ 'align' ] ) }
```
Comment (youknowriad) [inline packages/block-library/src/archives/edit.js:43]: What's the reason behind this change?

Comment (Addison-Stavlo) [reply packages/block-library/src/archives/edit.js:43]: Thanks for dropping in so quickly!

This is probably not the best solution, but it seems that removing the `align` attribute from block.json causes the SSR components request to 400:

Screen Shot 2020-07-28 at 12 07 02 PM

Since align is a saved attribute (but no longer in block.json?) it seems to be seen as an invalid attribute.  Breaking the block:

Screen Shot 2020-07-28 at 12 09 57 PM

If we omit the align attribute, the block continues to work as expected.  I think using `omit` is a bit hacky, so maybe we don't want to do it this way.  Perhaps this is why the attributes weren't removed previously? 🤔

File: packages/block-library/src/archives/index.js (hunk 2 above)
Comment (Addison-Stavlo) [reply packages/block-library/src/archives/index.js]: Its not necessary... we were misinterpreting  the error and I pushed that commit so @copons could take a look. 😄 

The deprecation is not necessary and is removed, the error stems from the other comment below regarding `omit` in the `ServerSideRender` component.

File: packages/block-library/src/archives/edit.js (hunk 3 above)
Comment (youknowriad) [reply packages/block-library/src/archives/edit.js:43]: Where's the "align" attribute registration happening now. It seems that the server-side registration is not picking up that attribute definition?

Comment (youknowriad) [reply packages/block-library/src/archives/edit.js:43]: I guess what I'm saying is that the `addAttribute` filter needs to move to the server now. We probably need to keep in both places though.

Comment (youknowriad) [reply packages/block-library/src/archives/edit.js:43]: I'm pretty sure the other attributes registered using "hooks" (color, lineHeight...) suffer the same issue?

Comment (Addison-Stavlo) [reply packages/block-library/src/archives/edit.js:43]: > I guess what I'm saying is that the addAttribute filter needs to move to the server now. We probably need to keep in both places though.

Yeah, that makes sense.  I wonder where this should be handled?  

> I'm pretty sure the other attributes registered using "hooks" (color, lineHeight...) suffer the same issue?

I would assume so, but I haven't run across any similar issues with them yet.

Im thinking it might make sense to do this in a future PR as it may inflate a bit from the 'simple cleanup' we were going for here.  While #24122 allows us to clean up some stuff in the index.php's here, this blocker for removing the defined align attribute for blocks using the SSR component seems to be a separate issue that will need to be addressed for all of the hooks.  So for these blocks I assume leaving their attributes as they were would be preferable to using the `omit` function?

Comment (Addison-Stavlo) [reply packages/block-library/src/archives/edit.js]: Updated to not remove the align attribute for blocks currently using the SSR component (no more `omit`, and their cleanup is limited to their index.php).  These blocks are:
* Archives
* Calendar
* Latest Comments
* RSS
* Tag Cloud

The following blocks do not use the SSR component and seem to have no issue with their align attribute being removed:
* Categories
* Latest Posts
* Navigation
* Search

**Review by Copons — APPROVED** [review]

File: packages/e2e-tests/fixtures/blocks/core__button__squared.json [hunk 4]
```diff
@@ -11,8 +11,7 @@
                     "text": "#1b9b6c",
                     "background": "#aa5a20"
                 }
-            },
-            "align": "none"
```
Comment (Addison-Stavlo) [inline packages/e2e-tests/fixtures/blocks/core__button__squared.json:15]: Since the Button component uses the align hook, and the align hook now has the enum for accepted values (right, center, left, wide, full), the test for `core__button__squared` was failing due to not returning `"align": "none"` as the fixture expected.  I removed this attribute from the fixture files for this test.  Hopefully this makes sense? 🤔

Comment (Addison-Stavlo) [reply packages/e2e-tests/fixtures/blocks/core__button__squared.json:15]: Or would it make more sense to leave this attribute in `core__button__squared.html` and `core__button__squared.parsed.json`, while only removing it from `core_button__squared.json` and `core__button__squared.serialized.html` (as opposed to just removing it from all 4)?  The idea being that if the attribute were declared in this way, we would expect it to not be returned according to the behavior of the align hook.

Comment (noahtallen) [reply packages/e2e-tests/fixtures/blocks/core__button__squared.json:15]: My opinion is that it makes sense to remove from all 4. I think having it in some and not others could be confusing. Plus, we aren't directly trying to test the "align" hook here, so if we do want to test that behavior, shouldn't it go somewhere else?

I might be missing something here, though, let me know!

File: packages/block-editor/src/hooks/align.js [hunk 5]
```diff
@@ -94,6 +94,9 @@ export function addAttribute( settings ) {
 			...settings.attributes,
 			align: {
 				type: 'string',
+				// Allow for '' since it is used by updateAlignment function
+				// in withToolbarControls for special cases with defined default values.
+				enum: [ ...ALL_ALIGNMENTS, '' ],
```
Comment (Addison-Stavlo) [inline packages/block-editor/src/hooks/align.js:99]: Updated this - It seems we need to allow the empty string as a value as it is used for special cases when default values are defined:
https://github.com/WordPress/gutenberg/blob/74c9065935fe995fba06f9c21ef9fb09f06dd6b2/packages/block-editor/src/hooks/align.js#L140-L142
